packages/components/src/range-control/utils.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { clamp, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useRef, useEffect, useState } from '@wordpress/element';
+
+/**
+ * A float supported clamp function for a specific value.
+ *
+ * @param {number} value The value to clamp
+ * @param {number} min The minimum value
+ * @param {number} max The maxinum value
+ * @return {number} A (float) number
+ */
+function floatClamp( value, min, max ) {
+	return parseFloat( clamp( value, min, max ) );
+}
+
+/**
+ * Hook to store a clamped value, derived from props.
+ */
+export function useControlledRangeValue( { min, max, value: valueProp = 0 } ) {
+	const [ value, _setValue ] = useState( floatClamp( valueProp, min, max ) );
+	const valueRef = useRef( value );
+
+	const setValue = useCallback(
+		( nextValue ) => {
+			_setValue( floatClamp( nextValue, min, max ) );
+		},
+		[ _setValue, min, max ]
+	);
```

Comment:
Is there any benefit to using `useCallback` here, vs. just a plain function?

Related: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Good catch! I don't think there's a need to over-optimize here. KCD saves the day again :)